### ISSUE TITLE
cli: Unhelpful cli/error.go:112  invalid arguments

### ISSUE THREAD

**Issue by kedare** [cockroachdb/cockroach#27698]
**BUG REPORT**

- What did you do?

Trying to start the cockroachDB container using this :
```
docker run -ti -v $(pwd):/var/lib/cockroach/ --rm cockroachdb/cockroach:v2.0.4 start --attrs environment:test --certs-dir /var/lib/cockroach/certs --http-port 8843 --listening-url-file /var/lib/cockroach/connection-string.txt --locality dc=bcn6 --max-disk-temp-storage 1G --sql-audit-dir /var/lib/cockroach/audir --store /var/lib/cockroach/data,size=50% --temp-dir /var/lib/cockroach/temp --logtostderr INFO
```

- What did you expect to see?
As there is probably a wrong argument key or value, a message telling me which one is wrong

- What did you see instead?

The full `--help` output following by an unhelpful `E180718 12:39:34.919239 1 cli/error.go:112  invalid arguments`

```
docker run -ti -v $(pwd):/var/lib/cockroach/ --rm cockroachdb/cockroach:v2.0.4 start --attrs environment:test --certs-dir /var/lib/cockroach/certs --http-port 8843 --listening-url-file /var/lib/cockroach/connection-string.txt --locality dc=bcn6 --max-disk-temp-storage 1G --sql-audit-dir /var/lib/cockroach/audir --store /var/lib/cockroach/data,size=50% --temp-dir /var/lib/cockroach/temp --logtostderr INFO
Usage:
  cockroach start [flags]

Examples:
  cockroach start --insecure --store=attrs=ssd,path=/mnt/ssd1 [--join=host:port,[host:port]]

Flags:
      --advertise-host string
        The hostname to advertise to other CockroachDB nodes for intra-cluster
        communication; it must resolve from other nodes in the cluster.

      --attrs string
        An ordered, colon-separated list of node attributes. Attributes are
        arbitrary strings specifying machine capabilities. Machine capabilities
        might include specialized hardware or number of cores (e.g. "gpu",
        "x16c"). For example:

          --attrs=x16c:gpu

      --background
        Start the server in the background. This is similar to appending "&"
        to the command line, but when the server is started with --background,
        control is not returned to the shell until the server is ready to
        accept requests.

      --cache bytes
        Total size in bytes for caches, shared evenly if there are multiple
        storage devices. Size suffixes are supported (e.g. 1GB and 1GiB). If
        left unspecified, defaults to 128MiB. A percentage of physical memory
        can also be specified (e.g. .25).
        (default 128 MiB)
      --certs-dir string
        The path to the directory containing SSL certificates and keys.

        Cockroach looks for certificates and keys inside the directory using the
        following naming scheme:

          - CA certificate and key: ca.crt, ca.key
          - Server certificate and key: node.crt, node.key
          - Client certificate and key: client.<user>.crt, client.<user>.key

        When running client commands, the user can be specified with the --user flag.
        Keys have a minimum permission requirement of 0700 (rwx------).
        This restriction can be disabled by setting the environment variable
        COCKROACH_SKIP_KEY_PERMISSION_CHECK to true.
        (default "${HOME}/.cockroach-certs")
      --external-io-dir string
        The local file path under which remotely-initiated operations that
        can specify node-local I/O paths, such as BACKUP, RESTORE or IMPORT,
        can access files. Following symlinks _is_ allowed, meaning that other
        paths can be added by symlinking to them from within this path.
        Note: operations in a distributed cluster can run across many nodes,
        so reading or writing to any given node's local file system in a
        distributed cluster is not usually useful unless that filesystem is
        actually backed by something like NFS.  If left empty, defaults to
        the "extern" subdirectory of the first store directory.  The value
        "disabled" will disable all local file I/O.

  -h, --help                          help for start
      --host string
        The hostname to listen on. The node will also advertise itself using
        this hostname if advertise-host is not specified.

      --http-host string
        The hostname or IP address to bind to for HTTP requests.

      --http-port string
        The port to bind to for HTTP requests.
        (default "8080")
      --insecure
        Start an insecure node, using unencrypted (non-TLS) connections,
        listening on all IP addresses (unless --host is provided) and disabling
        password authentication for all database users. This is strongly
        discouraged for production usage and should never be used on a public
        network without combining it with --host.

  -j, --join string
        The addresses for connecting a node to a cluster.

        When starting a multi-node cluster for the first time, set this flag to
        the addresses of 3-5 of the initial nodes. Then run the cockroach init
        command against one of the nodes to complete cluster startup.

        When starting a singe-node cluster, leave this flag out. This will
        cause the node to initialize a new single-node cluster without needing
        to run the cockroach init command.

        When adding a node to an existing cluster, set this flag to 3-5 of
        the nodes already in the cluster; it's easiest to use the same list of
        addresses that was used to start the initial nodes.

        This flag can be specified separately for each address:

          --join=localhost:1234 --join=localhost:2345

        Or can be specified as a comma separated list in single flag, or both
        forms can be used together, for example:

          --join=localhost:1234,localhost:2345 --join=localhost:3456



      --listening-url-file string
        After the CockroachDB node has started up successfully, it will write
        its connection URL to the specified file.

      --locality Locality
        An ordered, comma-separated list of key-value pairs that describe the
        topography of the machine. Topography might include country, datacenter
        or rack designations. Data is automatically replicated to maximize
        diversities of each tier. The order of tiers is used to determine
        the priority of the diversity, so the more inclusive localities like
        country should come before less inclusive localities like datacenter.
        The tiers and order must be the same on all nodes. Including more tiers
        is better than including fewer. For example:

          --locality=country=us,region=us-west,datacenter=us-west-1b,rack=12
          --locality=country=ca,region=ca-east,datacenter=ca-east-2,rack=4

          --locality=planet=earth,province=manitoba,colo=secondary,power=3
        (default )
      --max-disk-temp-storage bytes
        Maximum storage capacity available to store temporary disk-based
        data for SQL queries that exceed the memory budget (e.g. join,
        sorts, etc are sometimes able to spill intermediate results to disk).
        Accepts numbers interpreted as bytes, size suffixes (e.g. 32GB and
        32GiB) or a percentage of disk size (e.g. 10%). If left unspecified,
        defaults to 32GiB.  The location of the temporary files is within
        the first store dir (see --store). If expressed as a percentage,
        --max-disk-temp-storage is interpreted relative to the size of the
        storage device on which the first store is placed. The temp space usage
        is never counted towards any store usage (although it does share the
        device with the first store) so, when configuring this, make sure that
        the size of this temp storage plus the size of the first store don't
        exceed the capacity of the storage device. If the first store is an
        in-memory one (i.e. type=mem), then this temporary "disk" data is also
        kept in-memory. A percentage value is interpreted as a percentage of
        the available internal memory. If not specified, the default shifts to
        100MiB when the first store is in-memory.
        (default 0 B)
      --max-offset MaxOffset
        Maximum allowed clock offset for the cluster. If observed clock offsets
        exceed this limit, servers will crash to minimize the likelihood of
        reading inconsistent data. Increasing this value will increase the time
        to recovery of failures as well as the frequency of uncertainty-based
        read restarts.

        Note that this value must be the same on all nodes in the cluster.
        In order to change it, all nodes in the cluster must be stopped
        simultaneously and restarted with the new value.
        (default 500ms)
      --max-sql-memory bytes
        Maximum memory capacity available to store temporary data for SQL
        clients, including prepared queries and intermediate data rows during
        query execution. Accepts numbers interpreted as bytes, size suffixes
        (e.g. 1GB and 1GiB) or a percentage of physical memory (e.g. .25). If
        left unspecified, defaults to 128MiB.
        (default 128 MiB)
      --pid-file string
        After the CockroachDB node has started up successfully, it will write
        its process ID to the specified file.

  -p, --port string
        The port to bind to.
        (default "26257")
      --sql-audit-dir string
        If non-empty, create a SQL audit log in this drectory.
        (default "")
  -s, --store StoreSpec
        The file path to a storage device. This flag must be specified
        separately for each storage device, for example:

          --store=/mnt/ssd01 --store=/mnt/ssd02 --store=/mnt/hda1

        For each store, the "attrs" and "size" fields can be used to specify
        device attributes and a maximum store size (see below). When one or
        both of these fields are set, the "path" field label must be used for
        the path to the storage device, for example:

          --store=path=/mnt/ssd01,attrs=ssd,size=20GiB

        In most cases, node-level attributes are preferable to store-level
        attributes. However, the "attrs" field can be used to match
        capabilities for storage of individual databases or tables. For
        example, an OLTP database would probably want to allocate space for
        its tables only on solid state devices, whereas append-only time
        series might prefer cheaper spinning drives. Typical attributes include
        whether the store is flash (ssd), spinny disk (hdd), or in-memory
        (mem), as well as speeds and other specs. Attributes can be arbitrary
        strings separated by colons, for example:

          --store=path=/mnt/hda1,attrs=hdd:7200rpm

        The store size in the "size" field is not a guaranteed maximum but is
        used when calculating free space for rebalancing purposes. The size can
        be specified either in a bytes-based unit or as a percentage of hard
        drive space, for example:

          --store=path=/mnt/ssd01,size=10000000000     -> 10000000000 bytes
          --store=path=/mnt/ssd01,size=20GB            -> 20000000000 bytes
          --store=path=/mnt/ssd01,size=20GiB           -> 21474836480 bytes
          --store=path=/mnt/ssd01,size=0.02TiB         -> 21474836480 bytes
          --store=path=/mnt/ssd01,size=20%             -> 20% of available space
          --store=path=/mnt/ssd01,size=0.2             -> 20% of available space
          --store=path=/mnt/ssd01,size=.2              -> 20% of available space

        For an in-memory store, the "type" and "size" fields are required,
        and the "path" field is forbidden. The "type" field must be set to
        "mem", and the "size" field must be set to the true maximum bytes or
        percentage of available memory that the store may consume, for example:

          --store=type=mem,size=20GiB
          --store=type=mem,size=90%

        Commas are forbidden in all values, since they are used to separate
        fields. Also, if you use equal signs in the file path to a store, you
        must use the "path" field label.
        (default --store=path=/cockroach/cockroach-data)
      --temp-dir string
        The parent directory path where a temporary subdirectory will be
        created to be used for temporary files. This path must exist or the
        node will not start. The temporary subdirectory is used primarily
        as working memory for distributed computations and CSV importing.
        For example, the following will generate an arbitrary, temporary
        subdirectory "/mnt/ssd01/temp/cockroach-temp<NUMBER>":

          --temp-dir=/mnt/ssd01/temp

        If this flag is unspecified, the temporary subdirectory will be located
        under the root of the first store.

Global Flags:
      --log-backtrace-at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log-dir string                   if non-empty, write log files in this directory
      --log-dir-max-size bytes           maximum combined size of all log files (default 100 MiB)
      --log-file-max-size bytes          maximum size of each log file (default 10 MiB)
      --log-file-verbosity Severity      minimum verbosity of messages written to the log file (default INFO)
      --logtostderr Severity[=DEFAULT]   logs at or above this threshold go to stderr (default NONE)
      --no-color                         disable standard error log colorization
      --verbosity level                  log level for V logs
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging
E180718 12:39:34.919239 1 cli/error.go:112  invalid arguments
Error: invalid arguments
Failed running "start"
```

**Comment by knz** [comment]
@kedare thanks for your bug report. 

This situation has been fixed in CockroachDB 2.1; with the new version the program would properly indicate:

```
Error: unknown command "INFO" for "cockroach start"
Failed running "start"
```

And thereby reveal that the final argument should be `--logtostderr=INFO` instead of `--logtostderr INFO` (the log flags have a more constrained syntax, and arguably the help text is misleading. I filed this separately as #27834).

Does this clarify?
